### ISSUE TITLE
We need a full completed module for modern pages (Symfony) PS_1.7.5.1

### ISSUE THREAD

**Issue by Bossman1337** [PrestaShop/docs#211]
**Is your feature request related to a problem? Please describe.**
The current documentation isn't clear (https://devdocs.prestashop.com/1.7/development/architecture/migration-guide/forms/)

https://github.com/PrestaShop/PrestaShop/issues/11926 - This isn't the reply i would like either. A) That is for a legacy module B) The Main Forums seem slow/inactive C) Stackoverflow, no one replies as they don't know either.

**Describe the solution you'd like**
A full "Modern" module that we can download/install and view the source. A module that adds a new field to the product admin page. When the admin product page is loaded, it will load this new fields data, also when the product is saved it will save that field data. Would also be good to render the result to a twig template.

**Additional context**
The Prestashop documentation isn't the best (It's okay, but seems rushed tbh)

I'm developing/testing out creating new modules for PS. Symfony isn't the problem. Mainly its the best practices we need to learn and how Symfony and PS mix together, ready for current and future versions of PS. Currently from the dev docs it looks like we need to work in "src/PrestaShopBundle..." but how does that relate to a "Module", hows the install handled, or are all files meant to be within the module folder?

Basically, if a PS dev could knock up a quick module for us to download (Described above) it will answer 100's of questions in one go 👍

**Comment by Bossman1337** [comment]
@mickaelandrieu 

> But IMHO, it shouldn't be us (ie the PrestaShop company) but you the developers that should improve the documentation according to your needs and create this kind of skeletons 👍. We give you tools and you are free to adapt them to provide value to your customers, including creating your own skeletons that speed up your future developments!

I disagree with this statement. its like the chicken or egg question. How am i supposed to develop and improve the documentation if i don't know how to do it in the first place. PS is all about best practices etc. Take Symfony's documentation for example, this is exemplary and tonnes of it too with many examples.
If you Google anything regarding PS, it is very difficult to find anything up to date and is full of people just overriding instead of using hooks correctly. Trying to find anything regarding modern pages/modules is nonexistent.
Basically you're saying "Just look through all our code and work it out for yourself". You might like to work inefficiently, but we don't.

My next question to you would be: **Can** you or **can't** you create a "Modern module" like i described?

> We never say in the docs that people have to work in `src/`: would you mind to share with us where did you find such (wrong) advice?

Here: https://devdocs.prestashop.com/1.7/development/architecture/understanding-src-folder/
Here: https://devdocs.prestashop.com/1.7/development/architecture/migration-guide/ (the majority of the migration guide references the src.. folder) Its confusing tbh.

If you or any other dev that knows PS inside out, can create a module that i described then i'll create the documentation for it.

**Comment by Bossman1337** [comment]
@sarjon thanks for that, i will have a look. It still throws a lot of questions without commented code or documentation. Still think we need a module we can download and install, so we know it's complete and works.

**Comment by mickaelandrieu** [comment]
> Take Symfony's documentation for example, this is exemplary and tonnes of it too with many examples.

I can tell, to be one of the contributors that the Symfony documentation is mostly contributed by external contributors. You can see that when a community work together the result is great!

> If you Google anything regarding PS, it is very difficult to find anything up to date and is full of people just overriding instead of using hooks correctly.

I agree with you on this point: why PrestaShop developers don't share their knowledge with the community? I'm initially from the Symfony community (ex-SensioLabs employee, Symfony contributor, etc) and it's still a mystery for me. But a company "alone" can't create so many resources, not in a business model when our project is delivered for free to people.

> Here: https://devdocs.prestashop.com/1.7/development/architecture/understanding-src-folder/
> Here: https://devdocs.prestashop.com/1.7/development/architecture/migration-guide/ (the majority of > the migration guide references the src.. folder) Its confusing tbh.

The links you have visited are about the understanding of the complete architecture of PrestaShop and how people can help us to migrate the Back Office to Symfony, you will find information about the creation of modules in the module part of the documentation: https://devdocs.prestashop.com/1.7/modules/

Thanks @sarjon for sharing, can I fork your repository to put it into the *Friends of PrestaShop* organization? At some point, we could also accept it into the PrestaShop organization, but the core team won't have the time to maintain it so... ping @eternoendless 

I've just pushed a new module `customers` into the organization, but what is/are missing here is people writing articles, explaining what they have tested and learned, and able to contribute to the documentation.

We (the Core team) invest already a lot of resources into the documentation and the tests, but we won't be able to do all the work alone, we need all of you :+1:

**Comment by sarjon** [comment]
> Can I fork your repository to put it into the _Friends of PrestaShop_ organization

sure, it's open source. :+1: we could create some example module that does something useful with well commented code.

**Comment by Bossman1337** [comment]
@mickaelandrieu 

I'm looking over @sarjon module now. I think ultimately we need **"Modern Modules"** in the dev docs, the link you gave is for legacy module creation. I've created many modules this way and are fine. But as PS is going through a limbo period migrating to Symfony with only a few admin pages built with Symfony this https://devdocs.prestashop.com/1.7/modules/ doesn't explain database saving with Doctrine for example.

**Comment by frederic-benoist** [comment]
Hi,

I have put online some modules used during prestashop training.
You can find them on my github account. 

For the moment, the most interesting modules are only available during the training sessions. They should be posted online after the training materials have been updated.

**Comment by mickaelandrieu** [comment]
Note that I'm not against 1 page in the docs with a list of links about interesting modules, but I'm against "one more" module to be maintained by the Core team, we already maintain a lot (look at the number of active repositories in the PrestaShop organization: crazy, right?).

More, I repeat myself maybe, but I think that PrestaShop developers are more relevant to decide which "modern module" architecture is the best for their needs.

Of course, we will support any efforts :+1: 

Thanks for sharing @frederic-benoist !

**Comment by matks** [comment]
Hi, we have a released a demo module for 1.7.6 that uses:
- new grid (listing pages) hooks
- new identifiable object form (add/edit pages) hooks

This should cover a lot of your questions 😄 

https://github.com/friends-of-prestashop/demo-cqrs-hooks-usage-module/

**Comment by matks** [comment]
Closing the issue as I think https://github.com/friends-of-prestashop/demo-cqrs-hooks-usage-module is answering this need.

Feel free to open a new issue if there are topics uncovered 😉
